CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -621,6 +628,15 @@ private static void InlinesToHtml(HtmlTextWriter writer, Inline inline, CommonMa
                         stackWithinLink = withinLink;
                         break;
 
+                    case InlineTag.Placeholder:
+                        // the slim formatter will treat placeholders like literals, without applying any further processing
```

Comment:
I thought about that, but considered it untidy: Up to now, all steps preceding the creation of the AST are output-format-agnostic. It doesn't matter to them whether in the end, a formatter will produce HTML, Markdown again, RTF, PDF, or whatnot. The `CommonMarkSettings` are used in all stages, including those output-format-agnostic ones.

The return values of the placeholder resolver, on the other hand, may well be output-format-specific. Think of the aforementioned [Allura artifact links](https://sourceforge.net/p/forge/documentation/markdown_syntax/#md_ex_artlinks) - a placeholder such as `[#256]` will be resolved to a link such as `<a href="http://www.sourceforge.net/...=256">[#256]</a>`. A [table of contents placeholder in the same system - `[TOC]` -](https://sourceforge.net/p/forge/documentation/markdown_syntax/#md_ex_toc1), on the other hand, will be resolved to a structure of nested `<ul>` and `<li>` elements. In other words, the HTML elements have to be a part of what the placeholder resolver returns, and a placeholder resolver used for another output format will have to return something else instead.

Now, to enable placeholder resolution in the slim HTML formatter, I see two "tidy" options:

- Indeed add a placeholder resolver to the `CommonMarkSettings` class, but explicitly state that it is the one for the "default" output format, HTML, e.g. by naming it `HtmlPlaceholderResolver`. One needs to be aware, though, that possible formatters for non-HTML output formats will still have to keep their own placeholder resolver setting in the formatter itself, and that possible other slim formatters for non-HTML output formats may face the same dilemma of having no way to retrieve a placeholder resolver.
- Provide `CommonMarkSettings` with a mapping of output format IDs (e.g. MIME type strings?) to placeholder resolvers. While this would solve the aforementioned problems, this solution truly feels a bit like an overkill to me.

What do you think?